sklearn/model_selection/tests/test_validation.py

```diff
@@ -776,6 +777,44 @@ def split(self, X, y=None, groups=None):
 
     assert_raises(ValueError, cross_val_predict, est, X, y, cv=BadCV())
 
+    X, y = make_classification(n_classes=2, n_samples=50)
```

Comment:
Can we make this a separate function, like `test_cross_val_predict_decision_function_shape`?

Although I'd also like `predict_proba` similarly tested.